saleor/discount/utils.py

```diff
@@ -57,18 +56,17 @@ def get_product_discounts(product, discounts):
 def calculate_discounted_price(product, price, discounts):
     """Return minimum product's price of all prices with discounts applied."""
     if discounts:
-        discounts = list(
-            get_product_discounts(product, discounts))
+        discounts = list(get_product_discounts(product, discounts))
```

Comment:
Do we need to cast it to a list here?

Reply:
Without cast to list the `if discounts:` in next line will get generator and always evaluate to true and thus raise `min() arg is an empty sequence` if nothing is returned.